tests/lib.rs

```diff
@@ -1,149 +1,107 @@
+            "cirswgzloorzgc3djpjssazlwmvzhs5dinfxgoijbee======",
+        ),
+        (
+            Base32PadUpper,
+            "CIRSWGZLOORZGC3DJPJSSAZLWMVZHS5DINFXGOIJBEE======",
+        ),
+        (Base32HexLower, "v8him6pbeehp62r39f9ii0pbmclp7it38d5n6e89144"),
+        (Base32HexUpper, "V8HIM6PBEEHP62R39F9II0PBMCLP7IT38D5N6E89144"),
+        (Base32HexPadLower, "t8him6pbeehp62r39f9ii0pbmclp7it38d5n6e89144======"),
+        (Base32HexPadUpper, "T8HIM6PBEEHP62R39F9II0PBMCLP7IT38D5N6E89144======"),
+        (Base32Z, "het1sg3mqqt3gn5djxj11y3msci3817depfzgqejbrr"),
+        (Base58Flickr, "Z36tpRGiQ9Endr7dHahm9xwQdhmoER4emaRVT"),
+        (Base58Btc, "z36UQrhJq9fNDS7DiAHM9YXqDHMPfr4EMArvt"),
+        (Base64, "mRGVjZW50cmFsaXplIGV2ZXJ5dGhpbmchISE"),
+        (Base64Pad, "MRGVjZW50cmFsaXplIGV2ZXJ5dGhpbmchISE="),
+        (Base64Url, "uRGVjZW50cmFsaXplIGV2ZXJ5dGhpbmchISE"),
+        (Base64UrlPad, "URGVjZW50cmFsaXplIGV2ZXJ5dGhpbmchISE="),
+    ];
 
-    let (base, decoded) = decode("z1117paNL19xttacUY").unwrap();
-    assert_eq!(base, Base58btc);
-    assert_eq!(&decoded, id2)
+    for (base, output) in test_cases {
+        assert_eq!(encode(base, input), output);
+        assert_eq!(decode(output).unwrap(), (base, input.to_vec()));
+    }
 }
 
-
 #[test]
-fn test_decode() {
-    let id = b"Decentralize everything!!";
-
-    assert_eq!(
-        decode("f446563656e7472616c697a652065766572797468696e672121").unwrap(),
-        (Base16Lower, id.to_vec())
-    );
-
-    assert_eq!(
-        decode("f446563656e7472616c697a652065766572797468696e672121".to_string()).unwrap(),
-        (Base16Lower, id.to_vec())
-    );
-
-    assert_eq!(
-        decode("zUXE7GvtEk8XTXs1GF8HSGbVA9FCX9SEBPe").unwrap(),
-        (Base58btc, id.to_vec())
-    );
-
-    let id2 = b"yes mani !";
-
-    assert_eq!(
-        decode(
-            "011110010110010101110011001000000110110101100001011011100110100100100\
-             00000100001"
-        )
-        .unwrap(),
-        (Base2, id2.to_vec())
-    );
-    assert_eq!(
-        decode("7171312714403326055632220041").unwrap(),
-        (Base8, id2.to_vec())
-    );
-    assert_eq!(
-        decode("9573277761329450583662625").unwrap(),
-        (Base10, id2.to_vec())
-    );
-    assert_eq!(
-        decode("f796573206d616e692021").unwrap(),
-        (Base16Lower, id2.to_vec())
-    );
-    assert_eq!(
-        decode("Z7Pznk19XTTzBtx").unwrap(),
-        (Base58flickr, id2.to_vec())
-    );
-    assert_eq!(
-        decode("z7paNL19xttacUY").unwrap(),
-        (Base58btc, id2.to_vec())
-    );
-
-    assert_eq!(decode("mZg").unwrap(), (Base64UpperNoPad, b"f".to_vec()));
-    assert_eq!(decode("MZg==").unwrap(), (Base64UpperPad, b"f".to_vec()));
-    assert_eq!(decode("uZg").unwrap(), (Base64UrlUpperNoPad, b"f".to_vec()));
-    assert_eq!(decode("UZg==").unwrap(), (Base64UrlUpperPad, b"f".to_vec()));
+fn preserves_leading_zeroes() {
+    let input = b"\x00\x00\x00yes mani !";
+    println!("{:x?}", input.to_vec());
```

Comment:
Nitpick: this looks like a debug left-over.

(Nitpicks are things that I found, but are not worth changing if it is the only thing to change or if out feel like it's not worth the effort)